openmetadata-service/src/main/java/org/openmetadata/service/resources/datamodels/DataModelResource.java

```diff
@@ -0,0 +1,443 @@
+      throws IOException {
+    return dao.addFollower(securityContext.getUserPrincipal().getName(), id, userId).toResponse();
+  }
+
+  @DELETE
+  @Path("/{id}/followers/{userId}")
+  @Operation(
+      operationId = "deleteFollowerFromDataModel",
+      summary = "Remove a follower",
+      tags = "dataModels",
+      description = "Remove the user identified `userId` as a follower of the data model.")
+  public Response deleteFollower(
+      @Context UriInfo uriInfo,
+      @Context SecurityContext securityContext,
+      @Parameter(description = "Id of the data model", schema = @Schema(type = "UUID")) @PathParam("id") UUID id,
+      @Parameter(description = "Id of the user being removed as follower", schema = @Schema(type = "UUID"))
+          @PathParam("userId")
+          UUID userId)
+      throws IOException {
+    return dao.deleteFollower(securityContext.getUserPrincipal().getName(), id, userId).toResponse();
+  }
+
+  @DELETE
+  @Path("/{id}")
+  @Operation(
+      operationId = "deleteDataModel",
+      summary = "Delete a data model by `id`.",
+      tags = "dataModels",
+      description = "Delete a dataModel by `id`.",
+      responses = {
+        @ApiResponse(responseCode = "200", description = "OK"),
+        @ApiResponse(responseCode = "404", description = "DataModel for instance {id} is not found")
+      })
+  public Response delete(
+      @Context UriInfo uriInfo,
+      @Context SecurityContext securityContext,
+      @Parameter(description = "Hard delete the entity. (Default = `false`)")
+          @QueryParam("hardDelete")
+          @DefaultValue("false")
+          boolean hardDelete,
+      @Parameter(description = "Id of the data model", schema = @Schema(type = "UUID")) @PathParam("id") UUID id)
+      throws IOException {
+    return delete(uriInfo, securityContext, id, false, hardDelete);
+  }
+
+  @DELETE
+  @Path("/name/{fqn}")
+  @Operation(
+      operationId = "deleteDataModelByFQN",
+      summary = "Delete a data model by fully qualified name.",
+      tags = "dataModels",
+      description = "Delete a data model by `fullyQualifiedName`.",
+      responses = {
+        @ApiResponse(responseCode = "200", description = "OK"),
+        @ApiResponse(responseCode = "404", description = "DataModel for instance {fqn} is not found")
+      })
+  public Response delete(
+      @Context UriInfo uriInfo,
+      @Context SecurityContext securityContext,
+      @Parameter(description = "Hard delete the entity. (Default = `false`)")
+          @QueryParam("hardDelete")
+          @DefaultValue("false")
+          boolean hardDelete,
+      @Parameter(description = "Fully qualified name of the data model", schema = @Schema(type = "string"))
+          @PathParam("fqn")
+          String fqn)
+      throws IOException {
+    return deleteByName(uriInfo, securityContext, fqn, false, hardDelete);
+  }
+
+  @PUT
+  @Path("/restore")
+  @Operation(
```

Comment:
Do we need soft deleted and restore functionality for dataModels? cc @harshach

Reply:
Same here, I followed the same approach that we have for Charts.